thunder/executors/nvfuserex_impl.py

```diff
@@ -2194,3 +2194,41 @@ def map_inside_replacement(x: Any) -> None:
     elapsed_time_millis = elapsed_time_ns // 1000000
     rrctrace.set_provenance(TraceProvenance(f"Remove redundant casts (took {elapsed_time_millis} milliseconds)"))
     return rrctrace
+
+
+def _linear_check(a: TensorProxy, b: TensorProxy, bias: TensorProxy | None) -> bool:
+    if nv_version < LooseVersion("0.2.3"):
+        return False
+
+    enable_linear: None | bool = get_compile_option("nv_enable_linear", "Enable nvFuser matmul.")
+    if not enable_linear:
+        return False
+    # Verify linear inputs and bias (optional) are supported tensors.
+    if not are_supported_tensors(a, b):
+        return False
+    if bias is not None and not is_supported_tensor(bias):
```

Comment:
nitpick: We can probably return false in `is_supported_tensor(x)` when `x` is not a tensorproxy.

Reply:
I am not clear on this suggestion. We would still need to test for supported tensor only when bias is not None (which is a valid case).

Reply:
Ha, sorry I wasn't thinking straight. :+1: